crates/context_aware_config/src/api/dimension/types.rs

```diff
@@ -35,6 +36,36 @@ impl TryFrom<i32> for Priority {
     }
 }
 
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "Option<i32>")]
+pub struct Position(Option<i32>);
```

Comment:
Any reason for wrapping Option<i32> instead of i32 directly? You can do Option<Position> then, rather than deal with None and Some everywhere

Reply:
either case, it should be `Option<Position>`, where it is `Position(i64)`
because the parameter is optional, but currently it reads the parameter is mandatory which in turn takes optional value

Reply:
and when you would want to make it mandatory, you would want to change only the request type and not the properties of the type